12-offer-encoding.md

```diff
@@ -238,296 +233,328 @@ The human-readable prefix for offers is `lno`.
+        * [`...*chain_hash`:`chains`]
+    1. type: 4 (`offer_metadata`)
+    2. data:
+        * [`...*byte`:`data`]
+    1. type: 6 (`offer_currency`)
+    2. data:
+        * [`...*utf8`:`iso4217`]
+    1. type: 8 (`offer_amount`)
+    2. data:
+        * [`tu64`:`amount`]
+    1. type: 10 (`offer_description`)
+    2. data:
+        * [`...*utf8`:`description`]
+    1. type: 12 (`offer_features`)
+    2. data:
+        * [`...*byte`:`features`]
+    1. type: 14 (`offer_absolute_expiry`)
     2. data:
-        * [`sha256`:`offer_id`]
-    1. type: 8 (`amount`)
+        * [`tu64`:`seconds_from_epoch`]
+    1. type: 16 (`offer_paths`)
+    2. data:
+        * [`...*blinded_path`:`paths`]
+    1. type: 18 (`offer_issuer`)
+    2. data:
+        * [`...*utf8`:`issuer`]
+    1. type: 20 (`offer_quantity_min`)
+    2. data:
+        * [`tu64`:`min`]
+    1. type: 22 (`offer_quantity_max`)
+    2. data:
+        * [`tu64`:`max`]
+    1. type: 24 (`offer_node_id`)
+    2. data:
+        * [`point`:`node_id`]
+    1. type: 80 (`invoice_request_chain`)
+    2. data:
+        * [`chain_hash`:`chain`]
+    1. type: 82 (`invoice_request_amount`)
     2. data:
         * [`tu64`:`msat`]
-    1. type: 12 (`features`)
+    1. type: 84 (`invoice_request_features`)
     2. data:
         * [`...*byte`:`features`]
-    1. type: 32 (`quantity`)
+    1. type: 86 (`invoice_request_quantity`)
     2. data:
         * [`tu64`:`quantity`]
-    1. type: 38 (`payer_key`)
+    1. type: 88 (`invoice_request_payer_key`)
     2. data:
         * [`point`:`key`]
-    1. type: 39 (`payer_note`)
+    1. type: 89 (`invoice_request_payer_note`)
     2. data:
         * [`...*utf8`:`note`]
-    1. type: 50 (`payer_info`)
-    2. data:
-        * [`...*byte`:`blob`]
-    1. type: 56 (`replace_invoice`)
-    2. data:
-        * [`sha256`:`payment_hash`]
     1. type: 240 (`signature`)
     2. data:
         * [`bip340sig`:`sig`]
 
 ## Requirements for Invoice Requests
 
-The writer of an invoice_request:
-  - MUST set `payer_key` to a transient public key.
-  - MUST remember the secret key corresponding to `payer_key`.
-  - MUST set `offer_id` to the Merkle root of the offer as described in [Signature Calculation](#signature-calculation).
-  - MUST NOT set or imply any `chain_hash` not set or implied by the offer.
-  - MUST set `signature` `sig` as detailed in [Signature Calculation](#signature-calculation) using the `payer_key`.
-  - if the offer had a `quantity_min` or `quantity_max` field:
-    - MUST set `quantity`
+The writer:
+  - MUST copy all fields from the offer (including unknown fields).
+  - MUST NOT set any tlv fields greater or equal to 160.
+  - SHOULD set `invoice_request_payer_info` to an unpredictable series of bytes.
+  - MUST set `invoice_request_payer_key` to a transient public key.
+  - MUST remember the secret key corresponding to `invoice_request_payer_key`.
+  - MUST NOT set or imply any `invoice_request_chain` not set or implied by `offer_chains`.
```

Comment:
Unclear if not setting `invoice_request_chain ` implies (a) the chain set or implied by `offer_chains` or (b) bitcoin.

Reply:
Erk, yeach, this is a pain to specify :(

I've changed it to this:
```
  - if `offer_chains` is set:
    - MUST set `invoice_request_chain` to one of `offer_chains` unless that chain is bitcoin, in which case it MAY omit `invoice_request_chain`.
  - otherwise:
    - if it sets `invoice_request_chain` it MUST set it to bitcoin.
```